From ab9a76b59cd3dbd2051e7a447222d078f36d4d0f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Uwe=20Kleine-K=C3=B6nig?= <u.kleine-koenig@pengutronix.de>
Date: Fri, 8 Jan 2010 12:29:11 +0100
Subject: [PATCH 24/59] imx: remove unneeded include of mach/hardware.h
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This include isn't needed and hurts when defining static inline
functions in mach/hardware.h (or files included by mach/hardware.h).

The reason is that the consumers of mach/uncompress.h are compiled using
-Dstatic= so the code for all inline function is included in e.g.
arch/arm/boot/compressed/misc.o.

Signed-off-by: Uwe Kleine-König <u.kleine-koenig@pengutronix.de>
Cc: Sascha Hauer <kernel@pengutronix.de>
Cc: Russell King <linux@arm.linux.org.uk>
Cc: Jörg Knobloch <knobloch@incostartec.com>
Cc: Alan Carvalho de Assis <acassis@gmail.com>
Cc: Daniel Mack <daniel@caiaq.de>
Cc: Dmitriy Taychenachev <dimichxp@gmail.com>
---
 arch/arm/plat-mxc/include/mach/uncompress.h |    3 ---
 1 files changed, 0 insertions(+), 3 deletions(-)

Index: linux-2.6.33/arch/arm/plat-mxc/include/mach/uncompress.h
===================================================================
--- linux-2.6.33.orig/arch/arm/plat-mxc/include/mach/uncompress.h
+++ linux-2.6.33/arch/arm/plat-mxc/include/mach/uncompress.h
@@ -1,8 +1,6 @@
 /*
  *  arch/arm/plat-mxc/include/mach/uncompress.h
  *
- *
- *
  *  Copyright (C) 1999 ARM Limited
  *  Copyright (C) Shane Nay (shane@minirl.com)
  *
@@ -25,7 +23,6 @@
 
 #define __MXC_BOOT_UNCOMPRESS
 
-#include <mach/hardware.h>
 #include <asm/mach-types.h>
 
 static unsigned long uart_base;
